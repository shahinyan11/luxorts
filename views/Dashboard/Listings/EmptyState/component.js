import Image from 'next/image';
import Link from 'next/link';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import ROUTES from 'constants/routes';

import GradientButton from 'views/shared/GradientButton';

const EmptyState = ({ showNoResults, hasAppliedFilters, clearAllFilters }) => (
  <div className="listing__empty pt-64 pt-md-128 pb-64 pb-md-128">
    {showNoResults && (
      <>
        <div className="mb-24">
          <Image alt="" src="/images/no-results.png" width="136" height="136" />
        </div>
        <p className="listing__text mb-32">
          <FormattedMessage id="shared.noResultsFound" />
        </p>
        {hasAppliedFilters && (
          <GradientButton
            onClick={clearAllFilters}
            text={{ id: 'shared.clearAllFilters' }}
            className="min-width-140"
          />
        )}
      </>
    )}
    {!showNoResults && (
      <>
        <div className="mb-24">
          <Image alt="" src="/images/empty.png" width="136" height="136" />
        </div>
        <p className="listing__text mb-32">
          <FormattedMessage id="shared.noListingsYet" />
        </p>
        <Link href={ROUTES.ADD_NEW_LISTING.INDEX.PATH}>
          <a>
            <GradientButton text={{ id: 'shared.listYourProperty' }} className="min-width-180" />
          </a>
        </Link>
      </>
    )}
  </div>
);

EmptyState.propTypes = {
  clearAllFilters: PropTypes.func.isRequired,
  hasAppliedFilters: PropTypes.bool.isRequired,
  showNoResults: PropTypes.bool.isRequired,
};

export default EmptyState;
