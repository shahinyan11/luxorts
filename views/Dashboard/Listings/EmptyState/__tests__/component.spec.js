import { shallow } from 'enzyme';

import EmptyState from '../component';

describe('EmptyState component', () => {
  const props = {
    hasAppliedFilters: false,
    showNoResults: false,
    clearAllFilters: jest.fn(),
  };
  const component = shallow(<EmptyState {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when showNoResults equals true', () => {
    component.setProps({
      showNoResults: true,
    });

    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when showNoResults and hasAppliedFilters equal true', () => {
    component.setProps({
      showNoResults: true,
      hasAppliedFilters: true,
    });

    expect(component).toMatchSnapshot();
  });
});
