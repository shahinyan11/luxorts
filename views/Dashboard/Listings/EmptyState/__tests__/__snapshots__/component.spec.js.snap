// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState component matches snapshot 1`] = `
<div
  className="listing__empty pt-64 pt-md-128 pb-64 pb-md-128"
>
  <div
    className="mb-24"
  >
    <Image
      alt=""
      height="136"
      src="/images/empty.png"
      width="136"
    />
  </div>
  <p
    className="listing__text mb-32"
  >
    <Memo(FormattedMessage)
      id="shared.noListingsYet"
    />
  </p>
  <Link
    href="/add-new-listing"
  >
    <a>
      <GradientButton
        addonAfter={null}
        addonBefore={null}
        className="min-width-180"
        text={
          Object {
            "id": "shared.listYourProperty",
          }
        }
        variant="primary"
      />
    </a>
  </Link>
</div>
`;

exports[`EmptyState component matches snapshot when showNoResults and hasAppliedFilters equal true 1`] = `
<div
  className="listing__empty pt-64 pt-md-128 pb-64 pb-md-128"
>
  <div
    className="mb-24"
  >
    <Image
      alt=""
      height="136"
      src="/images/no-results.png"
      width="136"
    />
  </div>
  <p
    className="listing__text mb-32"
  >
    <Memo(FormattedMessage)
      id="shared.noResultsFound"
    />
  </p>
  <GradientButton
    addonAfter={null}
    addonBefore={null}
    className="min-width-140"
    onClick={[MockFunction]}
    text={
      Object {
        "id": "shared.clearAllFilters",
      }
    }
    variant="primary"
  />
</div>
`;

exports[`EmptyState component matches snapshot when showNoResults equals true 1`] = `
<div
  className="listing__empty pt-64 pt-md-128 pb-64 pb-md-128"
>
  <div
    className="mb-24"
  >
    <Image
      alt=""
      height="136"
      src="/images/no-results.png"
      width="136"
    />
  </div>
  <p
    className="listing__text mb-32"
  >
    <Memo(FormattedMessage)
      id="shared.noResultsFound"
    />
  </p>
</div>
`;
